internal/oci/runtime_oci.go

```diff
@@ -585,33 +587,34 @@ func waitContainerStop(ctx context.Context, c *Container, timeout time.Duration,
 			return fmt.Errorf("failed to wait process, timeout reached after %.0f seconds",
 				timeout.Seconds())
 		}
-		err := kill(c.state.Pid)
+		err := kill(c.Pid())
 		if err != nil {
 			return fmt.Errorf("failed to kill process: %v", err)
 		}
 	}
 
-	c.state.Finished = time.Now()
 	return nil
 }
 
 // StopContainer stops a container. Timeout is given in seconds.
-func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout int64) error {
+func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout int64) (errRet error) {
 	c.opLock.Lock()
-	defer c.opLock.Unlock()
+	defer func() {
+		c.opLock.Unlock()
+	}()
 
 	// Check if the process is around before sending a signal
-	process, err := findprocess.FindProcess(c.state.Pid)
+	process, err := findprocess.FindProcess(c.Pid())
```

Comment:
Get pid once and store it locally instead of multiple calls to c.Pid() in this function.